databus/src/main/java/com/bazaarvoice/emodb/databus/db/generic/CachingSubscriptionDAO.java

```diff
@@ -117,20 +122,24 @@ public CachingSubscriptionDAO(@CachingSubscriptionDAODelegate SubscriptionDAO de
                 .build(new CacheLoader<String, List<OwnedSubscription>>() {
                     @Override
                     public List<OwnedSubscription> load(String key) throws Exception {
-                        assert SUBSCRIPTIONS.equals(key) : "All subscriptions cache should only be accessed by a single key";
-
-                        Iterable<String> subscriptionNames = _delegate.getAllSubscriptionNames();
-                        ImmutableList.Builder<OwnedSubscription> subscriptions = ImmutableList.builder();
-
-                        for (String name : subscriptionNames) {
-                            // As much as possible take advantage of already cached subscriptions
-                            OwnedSubscription subscription = getSubscription(name);
-                            if (subscription != null) {
-                                subscriptions.add(subscription);
+                        try (Timer.Context ignore = _reloadAllSubscriptionsTimer.time()) {
+                            assert SUBSCRIPTIONS.equals(key) : "All subscriptions cache should only be accessed by a single key";
+
+                            Iterable<String> subscriptionNames = _delegate.getAllSubscriptionNames();
```

Comment:
should we have one more timer just on this call? never hurts. This helps specially if there is any issue with Line 133, the usage of cache for pulling up individual subscriptions.